### PR TITLE
feat(setup): add husky and lint-staged

### PR DESCRIPTION
**Pull request by grsabreu**
- [x] feature

Enforces every new commit to be linted with refmt

### CODE REVIEW

**Comment by grsabreu** [comment]
Closes https://github.com/apollographql/reason-apollo/issues/168

File: package.json [hunk 1]
```diff
@@ -45,8 +39,28 @@
...
+    "husky": "^1.2.0",
+    "lint-staged": "^8.1.0",
     "react-apollo": "^2.1.6",
     "reason-react": "^0.5.3"
   },
-  "dependencies": {}
+  "dependencies": {},
+  "lint-staged": {
+    "*.re": [
+      "refmt --in-place",
```
Comment (thangngoc89) [inline package.json]: you should use bsrefmt instead because it would work if bs-platform is installed

**Comment by Gregoirevda** [comment]
This was absolutely needed! Thanks

File: package.json (hunk 1 above)
Comment (Gregoirevda) [reply package.json]: what's the diff between refmt and bs-refmt?

Comment (thangngoc89) [reply package.json]: bsrefmt is just refmt but it's bundled with bs-platform,

**Pull request merged**